test.js

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import test from 'ava';
+import testUtil from './testutil';
+
+const projectRoot = process.cwd();
+const controlApp = path.join(projectRoot, 'control-app');
+const shrinkpackApp = path.join(projectRoot, 'shrinkpacked-app');
+const nonShrinkpackApp = path.join(projectRoot, 'non-shrinkpacked-app');
+const testPackageJson = path.join(projectRoot, 'testing/package.json');
+// const sinopiaConfigPath = path.join(projectRoot, 'testing/sinopia/config.yaml');
+// let killSinopia;
+
+test.before(async () => {
+  /* [killSinopia] = */await Promise.all([
+    // runSinopia(sinopiaConfigPath),
+    testUtil.setNpmVersion(process.env.NPM_VERSION),
+    testUtil.linkShrinkpack(projectRoot),
+    testUtil.createTestApp(controlApp, testPackageJson),
+    testUtil.createTestApp(shrinkpackApp, testPackageJson)
+  ]);
+  await testUtil.npmInstall(controlApp);
+  await testUtil.npmInstall(shrinkpackApp);
+});
+
+test.after.always(async () => {
```

Comment:
It might be worth caching the currently-installed `npm` version when the test run first starts, and then restoring it in this `after` block. Would make local testing less painful. Thoughts?